lab_1_keywords_tfidf/main.py

```diff
@@ -3,189 +3,238 @@
 
-    Args:
-        user_input (Any): Object to check
-        key_type (type): Expected type of dictionary keys
-        value_type (type): Expected type of dictionary values
-        can_be_empty (bool): Whether an empty dictionary is allowed
-
-    Returns:
-        bool: True if valid, False otherwise
-    """
+    # Args:
+    #     user_input (Any): Object to check
+    #     key_type (type): Expected type of dictionary keys
+    #     value_type (type): Expected type of dictionary values
+    #     can_be_empty (bool): Whether an empty dictionary is allowed
+
+    # Returns:
+    #     bool: True if valid, False otherwise
+    '''
+    if user_input == {}:
+        return can_be_empty
+    if isinstance(user_input, dict):
+        for key, value in user_input.items():
+            if not isinstance(key, key_type) or not isinstance(value, value_type):
+                return False
+        return True
+    return False
 
 
 def check_positive_int(user_input: Any) -> bool:
-    """
-    Check if the object is a positive integer (not bool).
+    '''
+    # Check if the object is a positive integer (not bool).
 
-    Args:
-        user_input (Any): Object to check
+    # Args:
+    #     user_input (Any): Object to check
 
-    Returns:
-        bool: True if valid, False otherwise
-    """
+    # Returns:
+    #     bool: True if valid, False otherwise
+    '''
+    if isinstance(user_input, int) and not isinstance(user_input, bool):
+        return user_input > 0
+    return False
 
 
 def check_float(user_input: Any) -> bool:
-    """
+    '''
     Check if the object is a float.
 
-    Args:
-        user_input (Any): Object to check
+    # Args:
+    #     user_input (Any): Object to check
 
-    Returns:
-        bool: True if valid, False otherwise
-    """
+    # Returns:
+    #     bool: True if valid, False otherwise
+    '''
+    return isinstance(user_input, float)
 
-
-def clean_and_tokenize(text: str) -> list[str] | None:
-    """
+def clean_and_tokenize(raw_text: str) -> list[str] | None:
+    '''
     Remove punctuation, convert to lowercase, and split into tokens.
 
-    Args:
-        text (str): Original text
-
-    Returns:
-        list[str] | None: A list of lowercase tokens without punctuation.
-        In case of corrupt input arguments, None is returned.
-    """
+    # Args:
+    #     text (str): Original text
+
+    # Returns:
+    #     list[str] | None: A list of lowercase tokens without punctuation.
+    #     In case of corrupt input arguments, None is returned.
+    '''
+    if isinstance(raw_text, str):
+        symbols_to_delete = '.,?-:;!%><#@$^&*()_'
+        raw_text = raw_text.replace('\n', ' ')
+        raw_text = raw_text.lower()
+        for symbol in raw_text:
+            if symbol in symbols_to_delete:
+                raw_text = raw_text.replace(symbol, '')
+        cleaned_tokens = raw_text.split()
+        return cleaned_tokens
+    return None
```

Comment:
you shoud minimize the code under if statement, so reverse if statement and everything will be good!